### PR TITLE
Updates the Fusion implementation to deal with the exponent properly when converting the hash to float

### PR DESCRIPTION
**Pull request by lightwolf**
I had to add the bit operators inline.

This fixes the issue where the fuse can't load the manifest because the hash to float conversion may create NaNs in some cases.

### CODE REVIEW

File: fusion/Modules/Lua/cryptomatte_utilities.lua [hunk 1]
```diff
@@ -56,6 +243,19 @@ function prefered_load(first, second)
...
+      for k,v in pairs(o) do
+         if type(k) ~= 'number' then k = '"'..k..'"' end
+         s = s .. '['..k..'] = ' .. dump(v) .. ','
+      end
+      return s .. '} '
+   else
+      return tostring(o)
+   end
+end
+
```
Comment (lightwolf) [inline fusion/Modules/Lua/cryptomatte_utilities.lua]: My bad, that can be removed and is just there for debugging puproses.

File: fusion/Modules/Lua/cryptomatte_utilities.lua [hunk 2]
```diff
@@ -868,9 +1068,16 @@ function CryptomatteInfo:parse_manifest(manifest)
     local all_names = {}
     local all_ids = {}
 
+    dump (bit)
```
Comment (lightwolf) [inline fusion/Modules/Lua/cryptomatte_utilities.lua]: Can be removed as well.

File: fusion/Modules/Lua/cryptomatte_utilities.lua [hunk 3]
```diff
@@ -4,11 +4,198 @@ Optional   : cjson
 Created by : Cédric Duriau         [duriau.cedric@live.be]
              Kristof Indeherberge  [xmnr0x23@gmail.com]
              Andrew Hazelden       [andrew@andrewhazelden.com]
-Version    : 1.2.4
+Version    : 1.2.5
```
Comment (EvanAW) [inline fusion/Modules/Lua/cryptomatte_utilities.lua:7]: Version number will need another update.

File: fusion/Modules/Lua/cryptomatte_utilities.lua [hunk 4]
```diff
@@ -4,11 +4,198 @@ Optional   : cjson
...
 --]]
 
 -- module table
 cryptomatte_utilities = {}
+--bit = require "bit"
+
+-----------------------
+-- bit lib implementions
```
Comment (cedricduriau) [inline fusion/Modules/Lua/cryptomatte_utilities.lua:15]: Could you link me towards the bit library you tried here?

File: fusion/Modules/Lua/cryptomatte_utilities.lua [hunk 5]
```diff
@@ -4,11 +4,198 @@ Optional   : cjson
...
+ end
+ return bit_and(n, 4294967295) -- 0xFFFFFFFF
+end
+
+function bit_xor2(m, n)
+ local rhs = bit_or(bit_not(m), bit_not(n))
+ local lhs = bit_or(m, n)
+ local rslt = bit_and(lhs, rhs)
+ return rslt
+end
```
Comment (cedricduriau) [inline fusion/Modules/Lua/cryptomatte_utilities.lua:196]: Considering the import statement above, and the completely different indentation, can I assume you extracted the code from an existing library? If my assumption is correct, this isn't really a kosher manner of re-using code, especially considering the open source standard nature of this repository. I do not mind using a third party when needed.

If my assumption was incorrect, my apologies.

File: fusion/Modules/Lua/cryptomatte_utilities.lua (hunk 4 above)
Comment (lightwolf) [reply fusion/Modules/Lua/cryptomatte_utilities.lua:15]: Good question, it's been a while.  The original link was http://luaforge.net/projects/bitlib but the domain is down. And the luaforge github (https://github.com/luaforge) doesn't include bitlib.

There's other repos using it though: https://github.com/q3k/hacklock/blob/master/software/main/lua-libs/bit.lua

**Comment by cedricduriau** [comment]
Hi @lightwolf 

Thanks a lot for taking the time to actually look into this and create the PR.

I realized that the Lua 5.2 `bit` module was backported to Lua 5.1 and is available builtin.
I was able to cover your implementation with that builtin module just fine.

Issue: https://github.com/cedricduriau/Cryptomatte/issues/14
PR: https://github.com/cedricduriau/Cryptomatte/pull/15

Cheers
Cedric

**Comment by lightwolf** [comment]
You're welcome. Sorry for the messy chode.

**Comment by cedricduriau** [comment]
> You're welcome. Sorry for the messy chode.

No need to apologize, I can only appreciate the time and effort you took to do this.

**Comment by jonahfriedman** [comment]
Thanks everyone!

**Pull request closed without merging**